lib/hammer_cli/ca_cert_fetcher.rb

```diff
@@ -58,6 +87,35 @@ def get_ca_cert(hostname, port)
       noverify_ssl_connection.peer_cert_chain.last
     end
 
+    def cert_hash(raw_cert)
+      cert = OpenSSL::X509::Certificate.new(raw_cert)
+      subject = OpenSSL::X509::Name.new(cert.subject)
+      subject.hash
+    end
+
+    def cert_file_name(ca_path, hostname)
+      File.join(ca_path, "#{hostname}.pem")
+    end
+
+    def store_ca_cert(raw_cert, ca_path, hostname)
+      cert_file = cert_file_name(ca_path, hostname)
+      File.write(cert_file, raw_cert)
+
+      # create link
+      hash = cert_hash(raw_cert)
+      cert_link = File.join(ca_path, "#{hash.to_s(16)}.%s")
+      count = 0
+      # increase hash index if file or link to different target already exist
+      while ((File.exist?(cert_link % count) || File.symlink?(cert_link % count)) &&
+          (File.symlink?(cert_link % count) == false ||
+            (File.symlink?(cert_link % count) &&
+            !(File.readlink(cert_link % count) =~ /\/#{hostname}.pem/)))) do
```

Comment:
Surprisingly file not exists? and is symlink? when the link points to non-existing location. Regardless I agree this needs to be simplified/refactored